### ISSUE TITLE
NativeColorBlock raises KeyError

### ISSUE THREAD

**Issue by lucasgueiros** [marteinn/wagtail-color-panel#10]
After adding a ColorBlock to a StreamField on wagtail page I got the following error on logs when I access the edit page:
```
Internal Server Error: /admin/pages/3/edit/preview/
Traceback (most recent call last):
  File "/app/venv/lib/python3.8/site-packages/django/utils/datastructures.py", line 76, in __getitem__
    list_ = super().__getitem__(key)
KeyError: 'secoes-count'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/app/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/app/venv/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/wagtail/admin/urls/__init__.py", line 170, in wrapper
    return view_func(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/wagtail/admin/auth.py", line 205, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/wagtail/admin/views/generic/preview.py", line 30, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/app/venv/lib/python3.8/site-packages/wagtail/admin/views/generic/preview.py", line 74, in post
    is_valid = form.is_valid()
  File "/app/venv/lib/python3.8/site-packages/wagtail/admin/forms/pages.py", line 176, in is_valid
    return super().is_valid()
  File "/app/venv/lib/python3.8/site-packages/modelcluster/forms.py", line 343, in is_valid
    form_is_valid = super().is_valid()
  File "/app/venv/lib/python3.8/site-packages/django/forms/forms.py", line 175, in is_valid
    return self.is_bound and not self.errors
  File "/app/venv/lib/python3.8/site-packages/django/forms/forms.py", line 170, in errors
    self.full_clean()
  File "/app/venv/lib/python3.8/site-packages/django/forms/forms.py", line 372, in full_clean
    self._clean_fields()
  File "/app/venv/lib/python3.8/site-packages/django/forms/forms.py", line 384, in _clean_fields
    value = field.widget.value_from_datadict(self.data, self.files, self.add_prefix(name))
  File "/app/venv/lib/python3.8/site-packages/wagtail/blocks/base.py", line 576, in value_from_datadict
    return self.block_def.value_from_datadict(data, files, name)
  File "/app/venv/lib/python3.8/site-packages/wagtail/blocks/stream_block.py", line 104, in value_from_datadict
    count = int(data["%s-count" % prefix])
  File "/app/venv/lib/python3.8/site-packages/django/utils/datastructures.py", line 78, in __getitem__
    raise MultiValueDictKeyError(key)
django.utils.datastructures.MultiValueDictKeyError: 'secoes-count'
```

My code is:

```python
@register_query_field("homepage", "homepages",query_params={"id": graphene.Int(), "slug": graphene.String()})
class HomePage(Page):
    # banners = models.ManyToManyField('home.Banner', null = True)
    secoes = StreamField([
        #("secao_imagem_em_cima",SecaoImagemEmCimaBlock()),
        ('title',blocks.CharBlock()),
        ('link', blocks.URLBlock()),
        ('texto', blocks.RichTextBlock()),
        ('icone', IconBlock()),
        ('cor', NativeColorBlock(default="#000000")),
        #('botao', ButtonBlock()),
    ], use_json_field = True)

    content_panels = Page.content_panels + [
        InlinePanel('banners'),
        FieldPanel('secoes'),
    ]
```

I am using:
```
Django==3.2.16
wagtail==4.1.1
wagtail-color-panel==1.4.0
```

**Comment by yahyaalbashar** [comment]
same problem here, any solution?

**Comment by marteinn** [comment]
Hi @lucasgueiros and @yahyaalbashar, the error is because the package is not yet compatible with Wagtail 4.1. It's in my to do list but I have no ETA for when it will be ready. Will feedback here once it's ready.

**Comment by yahyaalbashar** [comment]
> Hi @lucasgueiros and @yahyaalbashar, the error is because the package is not yet compatible with Wagtail 4.1. It's in my to do list but I have no ETA for when it will be ready. Will feedback here once it's ready.

Thanks for the quick response!
Just a note that I'm using wagtail 4.0.2 and the problem is appearing on the live server.

**Comment by marteinn** [comment]
Hi again! I'm having problems reproducing this issue, can you provide more details? Steps to reproduce would be most helpful. Thanks!

in my test i tried on existing projects running Wagtail 3 and Wagtail 4.1 and I can add the block to a streamfield and save the page without issues.

**Comment by lucasgueiros** [comment]
Hi. I am on holidays now. I will try to create a MRE next week and comment here. Thanks for attention.

**Comment by lucasgueiros** [comment]
Hi. I am on holidays now. I will try to create a MRE next week and comment here. Thanks for attention.
(Sorry, I cicked on close wrong)

**Comment by marts** [comment]
I've also got `wagtail-color-panel` on an existing project. I use `NativeColorBlock` and `ColorField` with `NativeColorPanel` and it seems to be working fine. I do have my `NativeColorBlock` as an item in a `blocks.StructBlock`. Don't know if that is significant. 

Django==3.2.16
wagtail==4.1.1
wagtail-color-panel==1.4.0
python==3.9.16

**Comment by lucasgueiros** [comment]
Hello. Sorry for the delay. I will post here the commands I executed to get the error. I am running on Ubuntu 22.04.1 LTS, python 3.10.6.

```bash
# Setting up env
sudo apt install python3.10-venv
mkdir teste-wagtail-color
cd teste-wagtail-color/
python3 -m venv ./venv
source venv/bin/activate

# Setup wagtail following https://docs.wagtail.org/en/stable/getting_started/tutorial.html
pip install wagtail
wagtail start teste_wagtail_color .
python manage.py migrate
python manage.py createsuperuser
pip install wagtail-color-panel
```

Here I edited home/models to contain the code below:

```python
from django.db import models
from wagtail.models import Page
from wagtail_color_panel.blocks import NativeColorBlock
from wagtail.fields import StreamField
from wagtail.admin.panels import FieldPanel

class HomePage(Page):
    secoes = StreamField([
        ("cor",NativeColorBlock()),
    ], null = True)
    content_panels = Page.content_panels + [
        FieldPanel("secoes"),
    ]
```

After editing I continued:
```bash
python manage.py makemigrations
python manage.py migrate
python manage.py runserver
```
Then I accessed [localhost/admin](http://localhost/admin) and navigated to editing Home page. When I click the pencil button it shows the same error:

![image](https://user-images.githubusercontent.com/9423116/214337999-38a6d4da-751b-4cc7-9f06-44f28706fb4c.png)

On logs I see this:
```
Internal Server Error: /admin/pages/3/edit/
Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292455479184

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292331193504

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292330289792

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292350389616

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292463368832

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 318, in build_node
    existing_node = self.nodes[obj_id]
KeyError: 140292350193248

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/template/backends/django.py", line 34, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/template/engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/template/engine.py", line 161, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: wagtail_color_panel/widgets/color-input-widget.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/views/decorators/cache.py", line 62, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/admin/urls/__init__.py", line 170, in wrapper
    return view_func(request, *args, **kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/admin/auth.py", line 205, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/admin/views/pages/edit.py", line 381, in dispatch
    return super().dispatch(request)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 142, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/admin/views/pages/edit.py", line 424, in get
    return self.render_to_response(self.get_context_data())
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/admin/views/pages/edit.py", line 873, in get_context_data
    + self.form.media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 196, in _media
    base = sup_cls.media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 196, in _media
    base = sup_cls.media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 196, in _media
    base = sup_cls.media
  [Previous line repeated 2 more times]
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/modelcluster/forms.py", line 355, in media
    media = super().media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 196, in _media
    base = sup_cls.media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/forms.py", line 495, in media
    media = media + field.widget.media
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/utils/functional.py", line 57, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/blocks/base.py", line 561, in media
    return self.js_context.media + forms.Media(
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/blocks/base.py", line 517, in js_context
    self._build_block_json()
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/blocks/base.py", line 512, in _build_block_json
    self._block_json = json.dumps(self._js_context.pack(self.block_def))
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 220, in pack
    return ValueContext(self).build_node(obj).emit()
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 339, in _build_new_node
    return adapter.build_node(obj, self)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 169, in build_node
    constructor, [context.build_node(arg) for arg in args]
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 169, in <listcomp>
    constructor, [context.build_node(arg) for arg in args]
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in _build_new_node
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in <listcomp>
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in _build_new_node
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in <listcomp>
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in _build_new_node
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 354, in <listcomp>
    return ListNode([self.build_node(item) for item in items])
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 339, in _build_new_node
    return adapter.build_node(obj, self)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 169, in build_node
    constructor, [context.build_node(arg) for arg in args]
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 169, in <listcomp>
    constructor, [context.build_node(arg) for arg in args]
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 321, in build_node
    node = self._build_new_node(val)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 339, in _build_new_node
    return adapter.build_node(obj, self)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 167, in build_node
    constructor, args = self.pack(obj, context)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/telepath/__init__.py", line 164, in pack
    return (self.js_constructor, self.js_args(obj))
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/widget_adapters.py", line 20, in js_args
    widget.render("__NAME__", None, attrs={"id": "__ID__"}),
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/utils/widgets.py", line 29, in render
    widget_html = self.render_html(name, value_data, attrs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/wagtail/utils/widgets.py", line 8, in render_html
    return super().render(name, value, attrs)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 281, in render
    return self._render(self.template_name, context, renderer)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/widgets.py", line 286, in _render
    return mark_safe(renderer.render(template_name, context))
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/renderers.py", line 28, in render
    template = self.get_template(template_name)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/forms/renderers.py", line 34, in get_template
    return self.engine.get_template(template_name)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/template/backends/django.py", line 36, in get_template
    reraise(exc, self)
  File "/home/lucas/Workspace/teste-wagtail-color/venv/lib/python3.10/site-packages/django/template/backends/django.py", line 85, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: wagtail_color_panel/widgets/color-input-widget.html
[24/Jan/2023 15:33:53] "GET /admin/pages/3/edit/ HTTP/1.1" 500 337602
```

**Comment by marteinn** [comment]
Hi @lucasgueiros: Question, has the app been added to installed apps? (https://github.com/marteinn/wagtail-color-panel/blob/main/docs/1_getting_started.md#quick-setup)

**Comment by marteinn** [comment]
Hi again @ucasgueiros, I just tried your MRE och a Wagtail 4.2 install and it worked well? I think your issue might be what I mentioned above, that the app is not among installed apps.

<img width="811" alt="Screenshot 2023-02-19 at 07 08 03" src="https://user-images.githubusercontent.com/286534/219932082-ca42bdde-3456-4976-bbc1-3ea21cb68d3b.png">

**Comment by lucasgueiros** [comment]
Hello. You are right about my MRE, I'm sorry about this silly mistake. I just forgot to verify if the error I found was the same that I reported on the issue, and it is not. My MRE says "TemplateDoesNotExist", but my original post says "django.utils.datastructures.MultiValueDictKeyError". I'm not currently using the plugin any more, I used another strategy that was even better for my use case. Unfortunately, I didn't commited when this error occurred and I lost that point in time. I suggest to wait to see if @yahyaalbashar can provide more details on the problem. Sorry for not being able to help anymore.
